File: 1753-maximum-score-from-removing-stones/1753-maximum-score-from-removing-stones.py
class Solution:
    def maximumScore(self, a: int, b: int, c: int) -> int:
        start = min(a,b,c)
        end = max(a,b,c)
        mid = (a+b+c) - (start + end)
        
        ans = 0
        while mid > 0 and end > 0:
            end -= 1
            mid -= 1
            total = start+mid+end
            start = min(start, mid, end)
            end = max(start, mid, end)
            mid = total - (start + end)
            ans += 1
        return ans

File: 1753-maximum-score-from-removing-stones/1753-maximum-score-from-removing-stones.cpp
class Solution {
public:
    int maximumScore(int a, int b, int c) {
        int ans = 0;
        priority_queue<int> pq; //maxheap
        pq.push(a);
        pq.push(b);
        pq.push(c);
        while(pq.size() > 1){
            ans++;
            int f = pq.top(); pq.pop();
            int s = pq.top(); pq.pop();
            if(f-1 > 0) pq.push(f-1);
            if(s-1 > 0) pq.push(s-1);
        }
        return ans;
    }
};